src/Operators/finitedifference.jl

```diff
@@ -2073,24 +2073,20 @@ Base.Broadcast._broadcast_getindex_eltype(
 ) = eltype(bc)
 
 # check that inferred output field space is equal to dest field space
-@noinline inferred_stencil_spaces_error(
-    dest_space_type::Type,
-    result_space_type::Type,
-) = error(
-    "dest space `$dest_space_type` is not the same instance as the inferred broadcasted result space `$result_space_type`",
+inferred_stencil_spaces_error(::Type{A}, ::Type{B}) where {A, B} = error(
+    "dest space `$A` is not the same instance as the inferred broadcasted result space `$B`",
 )
+inferred_stencil_spaces_error(::T, ::T) where {T} = nothing
 
 function Base.Broadcast.materialize!(
     ::Base.Broadcast.BroadcastStyle,
     dest::Fields.Field,
     bc::Base.Broadcast.Broadcasted{Style},
 ) where {Style <: AbstractStencilStyle}
     dest_space, result_space = axes(dest), axes(bc)
-    if result_space !== dest_space
-        # TODO: we pass the types here to avoid stack copying data
-        # but this could lead to a confusing error message (same space type but different instances)
-        inferred_stencil_spaces_error(typeof(dest_space), typeof(result_space))
-    end
+    # TODO: we pass the types here to avoid stack copying data
+    # but this could lead to a confusing error message (same space type but different instances)
+    inferred_stencil_spaces_error(dest_space, result_space)
```

Comment:
we don't want to pass the spaces in that was largely the point of above, otherwise it can lead to large stack copies.